### PR TITLE
SB-812 Stop loading classes unnecessarily

### PR DESCRIPTION
**Pull request by fuss86**

### CODE REVIEW

File: strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java [hunk 1]
```diff
@@ -17,20 +17,20 @@
 /**
  * @author Przemyslaw Fusik
  */
-public class AuthenticatorsClassLoader
+public class ExternalAuthenticatorsHelper
```
Comment (sbespalov) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]: Regarding [this](https://github.com/strongbox/strongbox/pull/278/files#diff-5007dae3de712baa3e32e2b397248eccL40), `WEB-INF/lib` is already in the `classpath`, so I suppose that we don't need this default value.
In general, I can't say that I like this reflection with `"addURL"` to add `Authenticator` jar files. I suggest you to use common `URLClassLoader` for this like below:
```
        ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
        URLClassLoader urlClassLoader = new URLClassLoader("path/to/authenticator/jar/files", currentClassLoader);
        Thread.currentThread().setContextClassLoader(urlClassLoader);
        //Load everything you need here
        Thread.currentThread().setContextClassLoader(currentClassLoader);
``` 
This looks clearly and will work independently of the environment (WAS, WebLogic. Jetty, Tomcat etc.).

**Comment by sbespalov** [comment]
And one more important thing about my [suggestion](https://drive.google.com/open?id=0ByK9S9e-c6SVdVpCZDExX3BpZjQ) above. In this case Authenticator classes will be loaded in child class loader which is important for isolation.

**Comment by fuss86** [comment]
@sbespalov, your link to suggestion points to some picture on google drive.

File: strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java (hunk 1 above)
Comment (fuss86) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]: I'll examine it later, thanks for your tips.

**Comment by sbespalov** [comment]
> @sbespalov, your link to suggestion points to some picture on google drive.
упс... :)
sorry for this, it's new fence in the cottage
link was updated

File: strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java (hunk 1 above)
Comment (carlspring) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]: @sbespalov / @fuss86 ,

After this proposed fix, I believe we can proceed with merging the pull, right?

Comment (sbespalov) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]: > After this proposed fix, I believe we can proceed with merging the pull, right?
yes

Comment (fuss86) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]: @sbespalov, I followed your suggestion, please review latest changes. 

Thanks.

Comment (fuss86) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]: @sbespalov, I followed your suggestion, please review latest changes. 

Thanks.

Comment (sbespalov) [inline strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java]: @fuss86 great! It looks better now )

**Comment by carlspring** [comment]
Great!

Should I go ahead and merge it?

On 3 Jun 2017 4:08 pm, "Sergey Bespalov" <notifications@github.com> wrote:

*@sbespalov* commented on this pull request.
------------------------------

In strongbox-user-management/src/main/java/org/carlspring/
strongbox/authentication/registry/support/ExternalAuthenticatorsHelper.java
<https://github.com/strongbox/strongbox/pull/278#discussion_r119988273>:

> @@ -17,20 +17,20 @@
 /**
  * @author Przemyslaw Fusik
  */
-public class AuthenticatorsClassLoader
+public class ExternalAuthenticatorsHelper

@fuss86 <https://github.com/fuss86> great! It looks better now )

—
You are receiving this because you were assigned.

Reply to this email directly, view it on GitHub
<https://github.com/strongbox/strongbox/pull/278#discussion_r119988273>,
or mute
the thread

**Comment by sbespalov** [comment]
> Should I go ahead and merge it?
@carlspring yes from me

**Comment by carlspring** [comment]
Thanks!

**Pull request merged**